packages/common/src/hooks/useSavedCollections.ts

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import { useDispatch, useSelector } from 'react-redux'
+
+import { Status } from 'models/Status'
+
+import { accountActions } from '../store/account'
+import {
+  savedCollectionsActions,
+  savedCollectionsSelectors
+} from '../store/saved-collections'
+
+const { fetchSavedPlaylists } = accountActions
+const { fetchCollections } = savedCollectionsActions
+
+const {
+  getAccountAlbums,
+  getSavedAlbumsState,
+  getAlbumsWithDetails,
+  getAccountPlaylists
+} = savedCollectionsSelectors
+
+const DEFAULT_PAGE_SIZE = 50
+
+export function useSavedAlbums() {
+  return useSelector(getAccountAlbums)
+}
+
+/* TODO: Handle filtering
+ * Option 1: This hook takes the list of album ids to fetch and computes the unfetched
+ * based on that.
+ * Option 2: Bake filter into selectors which drive this. Downside: Can't use this in multiple places...
+ */
+type UseSavedAlbumDetailsConfig = {
+  pageSize?: number
+}
+export function useSavedAlbumsDetails({
+  pageSize = DEFAULT_PAGE_SIZE
+}: UseSavedAlbumDetailsConfig) {
+  const dispatch = useDispatch()
+  const [hasFetched, setHasFetched] = useState(false)
+  const { unfetched: unfetchedAlbums, fetched: albumsWithDetails } =
+    useSelector(getAlbumsWithDetails)
+  const { status } = useSelector(getSavedAlbumsState)
+
+  const fetchMore = useCallback(() => {
+    if (status === Status.LOADING || unfetchedAlbums.length === 0) {
+      return
+    }
+    const ids = unfetchedAlbums
+      .slice(0, Math.min(pageSize, unfetchedAlbums.length))
```

Comment:
Doesn't the saga layer already chunk this via your `fetchCollectionChunks` fn?

Reply:
Yes, but there are two layers. This is intended for future use where we are calling `fetchMore` here and it's fetching the next X album records using the underlying saga. This has a configurable page size based on how many you want to fetch at once in a particular component.
The saga has no notion of page sizes. It's just a bulk fetch that guards against 400s by chunking up the list if you ask for too many at once.
For most purposes, the `pageSize` here will be much less than the chunk size in the saga. (i.e. we shoudn't ever trigger the saga's chunking.

Reply:
makes sense!